File: src/main/java/com/kapilkoju/autopos/service/impl/ItemServiceImpl.java
package com.kapilkoju.autopos.service.impl;

import com.kapilkoju.autopos.service.ItemService;
import com.kapilkoju.autopos.domain.Item;
import com.kapilkoju.autopos.repository.ItemRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * Service Implementation for managing Item.
 */
@Service
@Transactional
public class ItemServiceImpl implements ItemService{

    private final Logger log = LoggerFactory.getLogger(ItemServiceImpl.class);
    
    private final ItemRepository itemRepository;

    public ItemServiceImpl(ItemRepository itemRepository) {
        this.itemRepository = itemRepository;
    }

    /**
     * Save a item.
     *
     * @param item the entity to save
     * @return the persisted entity
     */
    @Override
    public Item save(Item item) {
        log.debug("Request to save Item : {}", item);
        Item result = itemRepository.save(item);
        return result;
    }

    /**
     *  Get all the items.
     *  
     *  @param pageable the pagination information
     *  @return the list of entities
     */
    @Override
    @Transactional(readOnly = true)
    public Page<Item> findAll(Pageable pageable) {
        log.debug("Request to get all Items");
        Page<Item> result = itemRepository.findAll(pageable);
        return result;
    }

    /**
     *  Get one item by id.
     *
     *  @param id the id of the entity
     *  @return the entity
     */
    @Override
    @Transactional(readOnly = true)
    public Item findOne(Long id) {
        log.debug("Request to get Item : {}", id);
        Item item = itemRepository.findOneWithEagerRelationships(id);
        return item;
    }

    /**
     *  Delete the  item by id.
     *
     *  @param id the id of the entity
     */
    @Override
    public void delete(Long id) {
        log.debug("Request to delete Item : {}", id);
        itemRepository.delete(id);
    }
}

File: src/main/kotlin/com/kapilkoju/autopos/catalog/service/ItemServiceImpl.kt
package com.kapilkoju.autopos.catalog.service


import com.kapilkoju.autopos.catalog.domain.Item
import org.slf4j.LoggerFactory
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

import org.springframework.transaction.annotation.Transactional


@Service
@Transactional
class ItemServiceImpl(private val itemRepo: ItemRepo)
    : ItemService {

    private val log = LoggerFactory.getLogger(ItemServiceImpl::class.java)


    override fun save(item: Item): Item {
        log.debug("Request to save Item : {}", item)
        return itemRepo.save(item)
    }


    @Transactional(readOnly = true)
    override fun findAll(query: String, pageable: Pageable): Page<Item> {
        log.debug("Request to get all Items")

        return if (query.isEmpty()) {
            itemRepo.findAll(pageable)
        } else {
            itemRepo.findByCodeIgnoreCaseContainingOrNameIgnoreCaseContaining(query, query, pageable)
        }
    }

    @Transactional(readOnly = true)
    override fun getItem(id: Long): Item? =
            itemRepo.findOne(id)

    @Transactional(readOnly = true)
    override fun getItemWithDetail(id: Long): Item? =
            itemRepo.findById(id)

    override fun delete(id: Long) {
        log.debug("Request to delete Item : {}", id)
        itemRepo.delete(id)
    }

    override fun adjustQuantity(item: Item, number: Int) {
        val dbItem = itemRepo.findOne(item.getId())
        dbItem.quantity = dbItem.quantity + number
        itemRepo.save(dbItem)
    }

}
